mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+"""
+This module provides pre-configured HTTP requests
+to consume a desired endpoint.
+"""
+
+from __future__ import annotations
+import sys
+import os
+import logging
+from copy import deepcopy
+from enum import Enum
+from requests import Request, Response, Session
+
+
+class Roles(Enum):
+    """
+    User roles available in McM
+    """
+
+    USER = "user"
+    GENERATOR_CONTACT = "generator_contact"
+    GENERATOR_CONVENER = "generator_convener"
+    PRODUCTION_MANAGER = "production_manager"
+    ADMINISTRATOR = "administrator"
+
+
+class APIRequest:
+    """
+    Provides some utilities to perform HTTP request for consuming
+    endpoints in the McM application impersonating a user or interacting
+    with the database.
+    """
+
+    def __init__(
+        self,
+        mcm_couchdb_url: str = "",
+        mcm_couchdb_credential: str = "",
+        mcm_couchdb_lucene_url: str = "",
+        mcm_application_url: str = "",
+        mockup: bool = False,
+    ) -> None:
+        """
+        Instantiate the API request client.
+
+        Attributes:
+            mcm_couchdb_url (str): McM database URL.
+                If it is not provided, it will be tried to retrieve it from
+                `MCM_COUCHDB_URL`.
+            mcm_couchdb_credential (str): Credentials to write data in
+                CouchDB. If it is not provided, it will be tried to retrieve it from
+                `COUCH_CRED`
```

Comment:
[Basic authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#basic_authentication)

Reply:
Related to the suggested change, the environment is queried only if the parameter was not provided when instantiating the object. Not sure if it is a good idea to keep it like this or maybe it is just better to remove this behavior.

Reply:
> [Basic authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#basic_authentication)

Sorry, I wanted to say that this should be documented.

> Related to the suggested change, the environment is queried only if the parameter was not provided when instantiating the object. Not sure if it is a good idea to keep it like this or maybe it is just better to remove this behavior.

I'm not sure, I think it would depend on the use cases we foresee. We could also have helper `@classmethod` `from_environment`

Reply:
Let's leave this as already is for the moment

Reply:
Solved